### PR TITLE
Dockerize Harvest and Registry Manager as a single image

### PR DESCRIPTION
**Pull request by ramesh-maddegoda**
## 🗒️ Summary
This pull request adds the Dockerfile, related files required to dockerize Harvest and Registry Manager as a single image.

## ♻️ Related Issues
Refer to the task https://github.com/NASA-PDS/pds-registry-app/issues/214

### CODE REVIEW

File: docker/build.sh [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    echo -e "\nThe file $PROPERTY_FILE does not exist."
+    echo -e "\tPlease provide an existing properties file as an argument to $0"
+    echo -e "\tUsage: $0 docker.properties \n"
+    exit 1
+fi
+
+# Returns the value for a given property key
+function getProperty {
+   PROPERTY_KEY=$1
+   PROPERTY_VALUE=$(cat "$PROPERTY_FILE" | grep -w "$PROPERTY_KEY" | cut -d'=' -f2)
```
Comment (nutjob4life) [inline docker/build.sh:63]: Note that any `cat FILE | COMMAND` can be replaced with `COMMAND < FILE` and reduce the process invocation count by one. You could also replace the `grep` and `cut` with `awk` but I'm too tired to figure that out this early in the morning 😇

File: docker/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+# ------------------------------------------------------------------------------
+# This shell script provides an entrypoint for the registry loader docker image.
+# ------------------------------------------------------------------------------
+
+# Check if the ES_URL environment variable is set
+if [ -z "$ES_URL" ]; then
+    echo "Error: 'ES_URL' (Elasticsearch URL) environment variable is not set. Use docker's -e option."
```
Comment (nutjob4life) [inline docker/entrypoint.sh:39]: This should be redirected to stderr
```
echo ERROR-MESSAGE 1>&2
```

File: docker/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    exit 1
+fi
+
+# Harvest data
+echo "Harvesting data ..."
```
Comment (nutjob4life) [inline docker/entrypoint.sh:44]: Similarly, these are diagnostic messages that should go to stderr (Unix philosophy: the output of any command may be the input to any other command; diagnostic & progress messages need a separate channel).

File: docker/entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+harvest -c "$HARVEST_CFG_FILE"
+
+# Load data into Elasticsearch
+echo "Loading harvested data into Elasticsearch ..."
```
Comment (nutjob4life) [inline docker/entrypoint.sh:48]: Ditto

File: docker/build.sh [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+HARVEST_VERSION=$(getProperty "harvest.version")
+REG_MANAGER_VERSION=$(getProperty "reg.manager.version")
+DOCKER_IMAGE_NAME_TAG=$(getProperty "docker.image.name.tag")
+
+echo "HARVEST_VERSION  = $HARVEST_VERSION"
+echo "REG_MANAGER_VERSION = $REG_MANAGER_VERSION"
+echo "DOCKER_IMAGE_NAME_TAG = $DOCKER_IMAGE_NAME_TAG"
+
+# Execute Docker build with build arguments
+docker build --build-arg harvest_version="$HARVEST_VERSION" \
```
Comment (nutjob4life) [inline docker/build.sh:78]: I like to be explicit and say `docker image build` to help underscore the difference between containers and images but that's just my taste 😋

File: docker/build.sh [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+# Both Harvest and Registry Manager tools are included in this Registry Loader docker image.
+# This script reads required configurations from a properties file passed as an argument.
+#
+# Usage: ./build.sh docker.properties
+#
+# ----------------------------------------------------------------------------------------------
+
+# Check if a properties file name is provided as an argument to this script
+if [ -z "$1" ]; then
+  echo -e "\nPlease provide a properties file as an argument to $0"
```
Comment (nutjob4life) [inline docker/build.sh:45]: Usage messages should go to stderr; `echo USAGE-MSG 1>&2`

**Review by nutjob4life — APPROVED** [review]
👍

File: docker/build.sh (hunk 1 above)
Comment (ramesh-maddegoda) [reply docker/build.sh]: Agreed. Replaced cat FILE | COMMAND with COMMAND < FILE.

File: docker/entrypoint.sh (hunk 2 above)
Comment (ramesh-maddegoda) [reply docker/entrypoint.sh:39]: Agreed. Redirected diagnostic messages and progress messages to stderr.

File: docker/entrypoint.sh (hunk 3 above)
Comment (ramesh-maddegoda) [reply docker/entrypoint.sh:44]: Done.

File: docker/entrypoint.sh (hunk 4 above)
Comment (ramesh-maddegoda) [reply docker/entrypoint.sh:48]: Done.

File: docker/build.sh (hunk 5 above)
Comment (ramesh-maddegoda) [reply docker/build.sh]: Agreed. Used `docker image build` instead of `docker build`.

**Review by nutjob4life — APPROVED** [review]
This looks 🆒

**Comment by ramesh-maddegoda** [comment]
Improved the code to download test data from a URL.

**Review by tloubrieu-jpl — APPROVED** [review]
Good to go !

**Pull request merged**